open-class/docs/classes/assignment-branch-logic.md

```diff
@@ -10,244 +10,187 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
-that you show to the user or that you will need to ask a follow-up question for. Sometimes, this choice
-making is called branching logic. It's the kind of choice that you might think of making with a flowchart.
+Very simple interviews might just ask a user some questions and spit out some result. Most interviews, though, will want to change what the questions they ask and the results they show based on a user's answers.
+
+Our code has to make a choice and, until the AI take over, we have to give it very specific instructions about how to make that choice. That's often called *branching logic*. Probably because of trees.
 
-In today's class, we'll talk about how to write a Docassemble interview that achieves this goal.
+## Example 1
 
-## The problem
+Suppose you have a friend who comes to you for advice. They were invited to a party, but they're not sure if they should go. You give them advice, your friend is happy, and now you have lots of friends, and even strangers coming to you looking for advice on the same topic. You decide [code is the answer](https://xkcd.com/1319/) and you'll write a simple form to help them out.
 
-Suppose you have a friend who comes to you for advice. They were invited to a party, but they're not
-sure if they should go. You give them advice, your friend is happy, and now you have lots of friends coming
-to you looking for advice on the same topic. You decide to write a simple flowchart to help them out.
+You start simple.
 
 ### Thinking through outcomes
 
-What outcomes will our flowchart have?
+What outcomes will our flowchart have? 
 
 1. Go to the party
 2. Stay home
-3. Go, but don't stay too late
-
-### Gathering the facts
-
-What kind of information do we want to know to advise our friends? Let's think of some possible questions.
-
-* Do you like parties?
-* Is there anyone there who you want to meet?
-* How difficult is it to get to the party?
-* Do you have any big assignments or studying you should do?
-
-### Sorting our questions out
-
-Let's think of a logical order for our questions, and how they fit together.
-
-It might make sense to first learn a little about what the person prefers. It makes
-sense to choose a question order that is respectful of the user's time. Let's think of big
-questions up front that might reduce the number of unnecessary questions we have to ask later.
-
-A logical first question might be:
-
-1. Do you like parties?
-
-If they don't like parties at all, they can probably skip this one. But just in case, a good follow-up might be:
-
-* Is there anyone there who you want to meet?
-
-Followed by
-
-* Do you have any big assigments or studying to do?
-
-If they do like parties, we might ask:
-
-2. Do you have a big assignment due?
-
-## Map it out
-
-For a small problem like this, it might be helpful to write out a small flow chart or map of the logic. For a bigger problem, you might come up with a different way to organize it.
-
```

Comment:
Note:

First, this is an example of where I'd have to ask myself hard questions about content. Are flowcharts part of my curriculum? If so, have they been introduced before? Does this lesson absolutely have to be taught here? Could there be two separate pages for Assignment 3 (or two separated sections of the lesson) that are each dedicated to their own concept - one for branching code and one for the UML representation?

Second, when I code I experiment and iterate. I don't generally plan the flow. That might have pros and cons, but it's how I've always worked, even when I was learning. Trying to plan this thoroughly would stop any forward momentum. As a beginner, it was because every aspect of this activity was completely new. I couldn't plan because I didn't know what was happening or what could happen. I not only didn't know what tools I had, I didn't even understand what tools were. I hadn't built up any intuition for what comes next.

I'm trying to think of a good metaphor, so this one might not be the best - Say you wanted to become a doctor. My first lesson is to say, "A patient comes into the room. What questions are you going to ask them?" Where would start? How can you plan that encounter?

As a beginner, that kind of planning needed to happen later on when I knew a bit more.

Now that I know more, I do a lot of the kind of questions you have earlier - what questions do I need to ask? What different outcomes are possible? Diagrams, for me, still aren't part of my process. I use them more for describing a system I've already created, not for brainstorming a system I haven't made yet.